src/library_glfw.js

```diff
@@ -671,6 +672,64 @@ var LibraryGLFW = {
       win.scrollFunc = cbfun;
     },
 
+    setDropCallback: function(winid, cbfun) {
+      var win = GLFW.WindowFromId(winid);
+      if (!win) return;
+      win.dropFunc = cbfun;
+    },
+
+    onDrop: function(event) {
+      if (!GLFW.active.dropFunc) return;
+
+      event.preventDefault();
+
+      var filenames = allocate(new Array(event.dataTransfer.files.length*4), 'i8*', ALLOC_NORMAL);
+      var filenamesArray = [];
+      var count = event.dataTransfer.files.length;
+
+      // Read and save the files to emscripten's FS
+      // TODO: lazily load?
+      var written = 0;
+      for (var i = 0; i < count; ++i) {
+        (function(file) {
+          var reader = new FileReader();
+          reader.onload = function(e) {
+            var data = e.target.result;
+            var path = file.name; // TODO: to a new directory?
+            FS.writeFile(path, new Uint8Array(data), { encoding: 'binary' });
```

Comment:
Note that `new Uint8Array` creates a [view](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray#Description) on the  existing [`ArrayBuffer`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer) already passed in so I wouldn't expect it to fail here. Testing with `emcc test_glfw_dropfile.c -o test_glfw_dropfile.html -s USE_GLFW=3 -s TOTAL_MEMORY=16MB`, I was able to drop a 100 MB file in the test program and read it successfully. The failure may occur in C if the program tries to read it all into memory, etc. I'll have to see what I can do to trigger an error; if onload isn't called for all files it may never call the callback and leak the filenames. The [loadend](https://developer.mozilla.org/en-US/docs/Web/Events/loadend) event looks promising (called after error/abort/load).